application/common/models/User.php

```diff
@@ -196,44 +120,110 @@ private function validatePassword(): Closure
     {
         return function ($attributeName) {
 
-            if ( ! $this->hasPasswordAlready()) {
+            if ($this->password_id === null) {
                 $this->attemptPasswordMigration();
             }
 
-            if (! password_verify($this->password, $this->password_hash)) {
+            /** @var Password $currentPassword */
+            $currentPassword = $this->getPassword()->one() ?? new Password();
```

Comment:
see https://github.com/silinternational/idp-id-broker/commit/c9752a3c1979dea7158a1eb89d0b6ca8742343c5 for related update